test/missing/core_test.clj

```diff
@@ -522,3 +522,56 @@
     (is (= () (merge-sort [[] []])))
     (is (= '(1 2 3 4 5 6) (merge-sort colls)))
     (is (= '(6 5 4 3 2 1) (merge-sort (flip compare) (map reverse colls))))))
+
+(deftest when-some-test
+  (let [atom (atom {})]
+    (when-some* [a 1
```

Comment:
You don't really need to use an atom for these tests, can just assert the return value.